lib/IDE/Refactoring.cpp

```diff
@@ -2602,6 +2602,108 @@ bool RefactoringActionLocalizeString::performChange() {
   return false;
 }
 
+void generateMemberwiseInit(SourceEditConsumer &EditConsumer,
+                            SourceManager &SM,
+                            SmallVector<std::string, 5>& memberNameVector,
+                            SmallVector<std::string, 5>& memberTypeVector,
+                            swift::SourceLoc targetLocation) {
+  
+  if (memberTypeVector.size() > 0) {
+    EditConsumer.accept(SM, targetLocation, "\ninternal init(");
+    
+    for (size_t i = 0; i < memberTypeVector.size(); i++) {
+      EditConsumer.accept(SM, targetLocation, memberNameVector[i] + ": " +
+                          memberTypeVector[i]);
+      
+      if (i != memberTypeVector.size() - 1) {
+        EditConsumer.accept(SM, targetLocation, ", ");
+      }
+    }
+    
+    EditConsumer.accept(SM, targetLocation, ") {\n\n");
+    
+    for (auto varName: memberNameVector) {
+      EditConsumer.accept(SM, targetLocation,
+                          "self." + varName + " = " + varName + "\n");
+    }
+    
+    EditConsumer.accept(SM, targetLocation, "\n}\n");
+  }
+}
+  
+ClassDecl* collectMembersForInit(ResolvedCursorInfo CursorInfo,
+                           SmallVector<std::string, 5>& memberNameVector,
+                           SmallVector<std::string, 5>& memberTypeVector) {
+  
+  if (CursorInfo.Kind != CursorInfoKind::ValueRef && !CursorInfo.ValueD)
+    return nullptr;
+  
+  ClassDecl *classDecl = dyn_cast<ClassDecl>(CursorInfo.ValueD);
+  if (!classDecl || classDecl->getStoredProperties().empty() ||
+      CursorInfo.IsRef) {
+    return nullptr;
+  }
+  
+  for (auto varDecl : classDecl->getStoredProperties()) {
+    auto parentPatternBinding = varDecl->getParentPatternBinding();
+    if (!parentPatternBinding) continue;
+    
+    auto varDeclIndex =
+    parentPatternBinding->getPatternEntryIndexForVarDecl(varDecl);
+    
+    if (auto init = varDecl->getParentPatternBinding()->getInit(varDeclIndex)) {
+      if (init->getStartLoc().isValid())
+        continue;
+    }
+    
+    StringRef memberName = varDecl->getName().str();
+    memberNameVector.push_back(memberName.str());
+    
+    std::string memberType = varDecl->getType().getString();
+    memberTypeVector.push_back(memberType);
+  }
+  
+  return classDecl;
+}
+
+bool RefactoringActionMemberwiseInitLocalRefactoring::
+isApplicable(ResolvedCursorInfo Tok, DiagnosticEngine &Diag) {
+  
+  SmallVector<std::string, 5> memberNameVector;
+  SmallVector<std::string, 5> memberTypeVector;
+  
+  auto classDecl = collectMembersForInit(Tok, memberNameVector,
+                                         memberTypeVector);
+  
+  return classDecl && memberTypeVector.size() > 0 &&
```

Comment:
We only need to check `!memberTypeVector .empty()` here to answer the `isApplicable` question.